### PR TITLE
Add MacOS plugin

### PR DESCRIPTION
**Pull request by ericwholt**
Add new MacOS plugin that parses `/var/log/system.log` and `/var/log/install.log`.

### CODE REVIEW

File: plugins/macos.yaml [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  - name: enable_system_log
+    label: System Logs
+    description: Enable to collect MacOS system logs
+    type: bool
+    default: true
+  - name: system_log_path
+    label: System Log Path
+    description: The absolute path to the System log
+    type: string
+    default: "/var/log/install.log"
```
Comment (jsirianni) [inline plugins/macos.yaml:15]: wrong path

File: plugins/macos.yaml [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  - name: enable_install_log
+    label: Install Logs
+    description: Enable to collect MacOS install logs
+    type: bool
+    default: true
+  - name: install_log_path
+    label: Install Log Path
+    description: The absolute path to the Install log
+    type: string
+    default: "/var/log/system.log"
```
Comment (jsirianni) [inline plugins/macos.yaml:25]: wrong path

File: plugins/macos.yaml [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    valid_values:
+      - beginning
+      - end
+    default: end
+
+# Set Defaults
+# {{$enable_system_log := default true .enable_system_log}}
+# {{$system_log_path := default "/var/log/system.log" .system_log_path}}
+# {{$enable_install_log := default true .enable_install_log}}
+# {{$install_log_path := default "/var/log/system.log" .install_log_path}}
```
Comment (jsirianni) [inline plugins/macos.yaml:39]: wrong path

File: plugins/macos.yaml [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+      plugin_id: {{ .id }}
+    output: system_parser
+
+  - id: system_parser
+    type: regex_parser
+    regex: '^(?P<timestamp>\w{3} \d{1,2} \d{2}:\d{2}:\d{2}) (---|(?P<host>[^ ]*))? ((?P<process>[^\[]*)\[(?P<pid>[^\]]*)\])?( \((?P<subprocess>[^\[]*)(\[(?P<spid>[^\]]*)\])?\))?(: )?(?P<message>[\w\W]*)'
+    timestamp:
+      parse_from: timestamp
+      layout_type: gotime
+      layout: 'Jan _2 15:04:05'
```
Comment (jsirianni) [inline plugins/macos.yaml:63]: Does this format always exist?

**Review by jsirianni — CHANGES_REQUESTED** [review]

File: plugins/macos.yaml (hunk 4 above)
Comment (ericwholt) [reply plugins/macos.yaml:63]: It existed in my log for a brief period of time in the middle of my log. It then switch back to timestamp format that seems to be the norm.  I am unsure it if this format shows up everywhere, but it caused errors with parsing my log file. It shouldn't hurt to handle it even it is just unique to my log.

File: plugins/macos.yaml (hunk 3 above)
Comment (ericwholt) [reply plugins/macos.yaml:39]: I will update

File: plugins/macos.yaml (hunk 2 above)
Comment (ericwholt) [reply plugins/macos.yaml:25]: I will update.

File: plugins/macos.yaml (hunk 1 above)
Comment (ericwholt) [reply plugins/macos.yaml:15]: I will update.

**Review by jsirianni — APPROVED** [review]

**Pull request merged**